Error `FreshClam exited with unexpected code: 14` in freshclam_update

Hello, I'm using this great construct in a stack, and the scanning functionality is working properly, but the definition update seems to be failing with an unexpected exit code. Looking at the man page for [freshclam](https://manpages.debian.org/unstable/clamav-freshclam/freshclam.1.en.html), this exit code is not documented.

```
[ERROR] Exception: 
{
    "source": "serverless-clamscan-update",
    "message": "FreshClam exited with unexpected code: 14"
}

Traceback (most recent call last):
  File "/var/lang/lib/python3.8/site-packages/aws_lambda_powertools/logging/logger.py", line 347, in decorate
    return lambda_handler(event, context)
  File "/var/task/lambda.py", line 38, in lambda_handler
    freshclam_update(download_path)
  File "/var/task/lambda.py", line 101, in freshclam_update
    report_failure(e.message)
  File "/var/task/lambda.py", line 111, in report_failure
    raise Exception(json.dumps(exception_json))
[ERROR] Exception: {"source": "serverless-clamscan-update", "message": "FreshClam exited with unexpected code: 14"} Traceback (most recent call last):   File "/var/lang/lib/python3.8/site-packages/aws_lambda_powertools/logging/logger.py", line 347, in decorate     return lambda_handler(event, context)   File "/var/task/lambda.py", line 38, in lambda_handler     freshclam_update(download_path)   File "/var/task/lambda.py", line 101, in freshclam_update     report_failure(e.message)   File "/var/task/lambda.py", line 111, in report_failure     raise Exception(json.dumps(exception_json))
```

Here is the more detailed output from the error: 

```
{
    "source": "serverless-clamscan-update",
    "message": "FreshClam exited with unexpected code: 14\nOutput: b\"ClamAV update process started at Tue Sep 14 00:19:57 2021\\ndaily database available for update (local version: 26291, remote version: 26293)\\nERROR: buildcld: Can't add daily.ldb to new daily.cld - please check if there is enough disk space available\\nERROR: updatedb: Incremental update failed. Failed to build CLD.\\nERROR: Unexpected error when attempting to update daily: Failed to update database\\nERROR: Database update process failed: Failed to update database\\nERROR: Update failed.\\n\""
}

Traceback (most recent call last):
  File "/var/lang/lib/python3.8/site-packages/aws_lambda_powertools/logging/logger.py", line 347, in decorate
    return lambda_handler(event, context)
  File "/var/task/lambda.py", line 38, in lambda_handler
    freshclam_update(download_path)
  File "/var/task/lambda.py", line 103, in freshclam_update
    report_failure(e.message)
  File "/var/task/lambda.py", line 113, in report_failure
    raise Exception(json.dumps(exception_json))
[ERROR] Exception: {"source": "serverless-clamscan-update", "message": "FreshClam exited with unexpected code: 14\nOutput: b\"ClamAV update process started at Tue Sep 14 00:19:57 2021\\ndaily database available for update (local version: 26291, remote version: 26293)\\nERROR: buildcld: Can't add daily.ldb to new daily.cld - please check if there is enough disk space available\\nERROR: updatedb: Incremental update failed. Failed to build CLD.\\nERROR: Unexpected error when attempting to update daily: Failed to update database\\nERROR: Database update process failed: Failed to update database\\nERROR: Update failed.\\n\""} Traceback (most recent call last):   File "/var/lang/lib/python3.8/site-packages/aws_lambda_powertools/logging/logger.py", line 347, in decorate     return lambda_handler(event, context)   File "/var/task/lambda.py", line 38, in lambda_handler     freshclam_update(download_path)   File "/var/task/lambda.py", line 103, in freshclam_update     report_failure(e.message)   File "/var/task/lambda.py", line 113, in report_failure     raise Exception(json.dumps(exception_json))
```

--- a/API.md
+++ b/API.md
@@ -23,7 +23,7 @@ An [aws-cdk](https://github.com/aws/aws-cdk) construct that uses [ClamAV®](http
 The construct creates a Lambda function with EFS integration to support larger files.
 A VPC with isolated subnets, a S3 Gateway endpoint will also be created.
 
-Additionally creates an hourly job to download the latest ClamAV definition files to the
+Additionally creates an twice-daily job to download the latest ClamAV definition files to the
 Virus Definitions S3 Bucket by utilizing an EventBridge rule and a Lambda function and
 publishes CloudWatch Metrics to the 'serverless-clamscan' namespace.
 

--- a/assets/lambda/code/download_defs/lambda.py
+++ b/assets/lambda/code/download_defs/lambda.py
@@ -43,7 +43,10 @@ def download_s3_defs(download_path, defs_bucket):
     """download CVD and conf files from definitions bucket (if they exist)
     to compare against ClamAV database. Respect their hosting costs!"""
     try:
-        file_regex = [r"\w+.c[vl]d", r"freshclam.conf"]
+        # Downloading ClamAV definitions and exceeds Lambda's tmp directory max size
+        # https://github.com/awslabs/cdk-serverless-clamscan/issues/118
+        # file_regex = [r"\w+.c[vl]d", r"freshclam.conf"]
+        file_regex = [r"freshclam.conf"]
         file_pattern = r"||".join(file_regex)
         for file in defs_bucket.objects.all():
             filename = file.key
@@ -78,6 +81,7 @@ def freshclam_update(download_path):
             f.write("\nDNSDatabaseInfo current.cvd.clamav.net")
             f.write("\nDatabaseMirror  database.clamav.net")
             f.write("\nReceiveTimeout  0")
+            f.write("\nCompressLocalDatabase  true")
     try:
         command = [
             "freshclam",

--- a/src/index.ts
+++ b/src/index.ts
@@ -92,7 +92,7 @@ export interface ServerlessClamscanProps {
   The construct creates a Lambda function with EFS integration to support larger files.
   A VPC with isolated subnets, a S3 Gateway endpoint will also be created.
 
-  Additionally creates an hourly job to download the latest ClamAV definition files to the
+  Additionally creates an twice-daily job to download the latest ClamAV definition files to the
   Virus Definitions S3 Bucket by utilizing an EventBridge rule and a Lambda function and
   publishes CloudWatch Metrics to the 'serverless-clamscan' namespace.
 
@@ -418,7 +418,7 @@ export class ServerlessClamscan extends Construct {
     defs_bucket.grantReadWrite(download_defs);
 
     new Rule(this, 'VirusDefsUpdateRule', {
-      schedule: Schedule.rate(Duration.hours(1)),
+      schedule: Schedule.rate(Duration.hours(12)),
       targets: [new LambdaFunction(download_defs)],
     });
 

